front-end/src/index.js

```diff
@@ -2,57 +2,42 @@ import React, { useState } from "react"
 import ReactDOM from "react-dom/client"
 import "./index.css"
 
-const axios = require('axios')
+import Head from "./components/Head/Head"
+import Footer from "./components/Footer/Footer"
+import Showurl from "./components/ShowUrl/Showurl"
 
+const axios = require('axios')
 
-function Main() {
+function App() {
 
     const [link, setLink] = useState('')
     const [url, setUrl] = useState('')
 
-    // async function requisitar() {
-    //     await axios.get('http://127.0.0.1:5000/requisitar').then(function (response) { 
-
-    //         setUrl(response.data.link)
-
-    //     }).catch(function (error) {
-    //         console.log(error)
-    //     })
-    // }
-
-
     async function save() {
 
         await axios.post('http://localhost:5000/savelink', JSON.parse(`{"link": "${link}" }`)).then((response) => {
```

Comment:
deveria mudar para receber o valor numa variável, fazendo melhor uso do async await. (JSON.parse() não é necessário pq o axios faz isso internamente)
```suggestion
        const { data } = await axios.post('http://localhost:5000/savelink', {link: link});
        setUrl(data.link);
        setLink('');
```

Reply:
Resolvido, assim fica até mais fácil.